### PR TITLE
Add micronaut-rest module

### LINKED ISSUE #32: Create a new module microstream-rest
### Issue description

It should have an `implementation` dependency to: 

https://github.com/microstream-one/microstream/blob/master/storage/rest/service-sparkjava/pom.xml#L26-L30

### PR DESCRIPTION
**Pull request by timyates**
Fixes #32

### CODE REVIEW

File: microstream-rest/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+plugins {
+    id("io.micronaut.build.internal.module")
+}
+
+dependencies {
+    implementation(libs.managed.microstream.storage.restservice)
+    testImplementation(project(":microstream"))
+    testImplementation(libs.logback.classic)
```
Comment (sdelamo) [inline microstream-rest/build.gradle.kts:8]: `testRuntimeOnly` ?

File: microstream-rest/src/main/java/io/micronaut/microstream/rest/StorageManagerListener.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Tim Yates
+ */
+@Singleton
+@Requires(bean = StorageManager.class)
+@Requires(property = "microstream.rest.enabled", value = StringUtils.TRUE, defaultValue = StringUtils.FALSE)
+public class StorageManagerListener implements BeanCreatedEventListener<StorageManager> {
```
Comment (sdelamo) [inline microstream-rest/src/main/java/io/micronaut/microstream/rest/StorageManagerListener.java:35]: what is this for?

File: microstream-rest/src/test/groovy/io/micronaut/microstream/rest/StorageManagerListenerFuncTest.groovy [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import ch.qos.logback.core.read.ListAppender
+import io.micronaut.context.ApplicationContext
+import io.micronaut.inject.qualifiers.Qualifiers
+import one.microstream.storage.types.StorageManager
+import org.slf4j.LoggerFactory
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.TempDir
+
+class StorageManagerListenerFuncTest extends Specification {
```
Comment (sdelamo) [inline microstream-rest/src/test/groovy/io/micronaut/microstream/rest/StorageManagerListenerFuncTest.groovy:14]: I think we should remove this test and the listener.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: microstream-rest/build.gradle.kts (hunk 1 above)
Comment (timyates) [reply microstream-rest/build.gradle.kts]: Gone now

File: microstream-rest/src/main/java/io/micronaut/microstream/rest/StorageManagerListener.java (hunk 2 above)
Comment (timyates) [reply microstream-rest/src/main/java/io/micronaut/microstream/rest/StorageManagerListener.java]: I was going the wrong way

Gone now

File: microstream-rest/src/test/groovy/io/micronaut/microstream/rest/StorageManagerListenerFuncTest.groovy (hunk 3 above)
Comment (timyates) [reply microstream-rest/src/test/groovy/io/micronaut/microstream/rest/StorageManagerListenerFuncTest.groovy]: done

**Comment by timyates** [comment]
Code smell will go once #37 is in

**Review by sdelamo — APPROVED** [review]

**Pull request merged**